### PR TITLE
feat(observability): add Prometheus metrics for BirdNET processing buffer overruns

### PR DESCRIPTION
**Pull request by tphakala**
## Summary

- Adds three new Prometheus metrics to `MyAudioMetrics` for tracking BirdNET processing buffer overruns (when inference time exceeds the effective buffer duration):
  - `myaudio_birdnet_processing_overruns_total` — counter per source
  - `myaudio_birdnet_processing_overrun_duration_seconds` — histogram of elapsed processing time
  - `myaudio_birdnet_processing_overrun_ratio` — histogram of elapsed/buffer ratio (>1.0 = overrun)
- Wires metric recording into the existing overrun detection in `ProcessData()`, complementing the Sentry telemetry added in #2229

## Test plan

- [x] Unit tests for `RecordBirdNETProcessingOverrun` (counter, histograms, zero buffer edge case)
- [x] `golangci-lint` passes on changed packages
- [x] `go test -race` passes

## Summary by CodeRabbit

* **New Features**
  * Added observability metrics for BirdNET audio processing buffer overruns: counts, duration histogram, and ratio histogram (custom buckets) plus a recording API.

* **Tests**
  * Added comprehensive tests verifying metrics recording, accumulation across events, and behavior when buffer length is zero.

### CODE REVIEW

File: internal/myaudio/process.go [hunk 1]
```diff
@@ -206,6 +206,14 @@ func ProcessData(bn *birdnet.BirdNET, data []byte, startTime time.Time, source s
 			logger.Duration("buffer_length", effectiveBufferDuration),
 			logger.String("source", source))
 		recordBufferOverrun(elapsedTime, effectiveBufferDuration)
+
+		// Record Prometheus metrics for observability
+		processMetricsMutex.RLock()
+		m := processMetrics
+		processMetricsMutex.RUnlock()
+		if m != nil {
+			m.RecordBirdNETProcessingOverrun(source, elapsedTime.Seconds(), effectiveBufferDuration.Seconds())
```
[the remark this answers is not in the document]
Comment (tphakala) [reply internal/myaudio/process.go:215]: The `source` parameter here is already a sanitized registry ID (e.g. `rtsp_c9562b17`), not a raw connection string. The call chain is: `analysis_buffer.go:660` calls `ProcessData(bn, data, startTime, sourceID)` where `sourceID` comes from the buffer's registry key. The same `source` value is used by pre-existing metrics like `RecordAnalysisBufferProcessingDuration` and in `log.Warn` calls. No cardinality or privacy concern here.

File: internal/observability/metrics/myaudio_test.go [hunk 2]
```diff
@@ -104,6 +104,59 @@ func BenchmarkRecordAudioConversionError_FmtSprintf(b *testing.B) {
...
+	})
+
+	t.Run("zero buffer length skips ratio", func(t *testing.T) {
+		source := "mic_zero"
+		// Should not panic with zero buffer length
+		m.RecordBirdNETProcessingOverrun(source, 3.0, 0)
+
+		count := testutil.ToFloat64(m.birdnetProcessingOverrunsTotal.WithLabelValues(source))
+		assert.InDelta(t, 1.0, count, 0.01)
+	})
```
[the remark this answers is not in the document]
Comment (tphakala) [reply internal/observability/metrics/myaudio_test.go]: Fixed in 75169c24 — each subtest now gets its own `prometheus.NewRegistry()` and `NewMyAudioMetrics` instance.
